### PR TITLE
Allow anonymous azure blob requests

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

**Review by Xuanwo — COMMENTED** [review]
Thanks for contribution! I left some reivew comments about it, PTAL.

File: src/azure/storage/signer.rs [hunk 1]
```diff
@@ -67,15 +67,19 @@ impl Builder {
     /// The builder should not be used anymore.
     pub fn build(&mut self) -> Result<Signer> {
         let mut cred_loader = CredentialLoader::default();
+        let mut allow_anonymous = false;
         if self.credential.is_valid() {
             cred_loader = cred_loader.with_credential(self.credential.clone());
+        } else {
+            // If the credential is not valid, assume we want anonymous access.
```
[the remark this answers is not in the document]
Comment (agocke) [reply src/azure/storage/signer.rs:74]: Makes sense. Add a bool property to the builder and let people set the value?

[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/azure/storage/signer.rs:74]: LGTM!

File: src/azure/storage/signer.rs [hunk 2]
```diff
@@ -66,20 +72,24 @@ impl Builder {
     ///
     /// The builder should not be used anymore.
     pub fn build(&mut self) -> Result<Signer> {
+        if self.use_anonymous {
```
Comment (Xuanwo) [inline src/azure/storage/signer.rs:75]: The API I want to expose to users is `allow_anonymous` which is different from `use_anonymous`.

- For `allow_anonymous`, users can fetch credentials from env, and if not found, allow visiting azblob as anonymous.
- For `use_anonymous`, users have to decide visiting azblob as anonymous or not during compilation which is less useful.

**Review by Xuanwo — APPROVED** [review]
Nice PR!

**Pull request merged**